### DISCUSSION TITLE
searchCertificates REST endpoint returns a trimmed serial number

### DISCUSSION THREAD

**Discussion by AbdouMechraoui** [Keyfactor/ejbca-ce#673] [Q&A] [answered]
Hi,

The serial number of a certificate returned by the REST endpoint `ejbca/ejbca-rest-api/v2/certificate/search`  is missing leading zero/s.
Interestingly, the leading zero is present in the web UI. 

Actual serial number (web UI): `0ABCDE622552BC2533C84208DA1AEC380D8454AC`

REST reposnse using cURL:
```json
{
    "serialNumber":"ABCDE622552BC2533C84208DA1AEC380D8454AC"
}
```

What is the reason for this discrepeny?

**Comment by primetomas** [comment, marked as answer]
Ah, there you see. THis is just a formatting thing. Some output, say by default Java BigInteger.toString(16) does not include leading zeroes. 
The leading zero does not contribute anything to the value. Hex converting it to decimal gives the same value regardless of the leading zero or not.

**Reply by AbdouMechraoui** [reply]
That makes sense! Thank Tomas :D
